service/equation.js

```diff
@@ -0,0 +1,96 @@
+import { ValidationError, UniqueConstraintError } from 'sequelize';
+
+import equationRepository from '../repository/equation';
+import {
+  InexistentItem,
+  CustomUniqueConstraintError,
+  CustomValidationError,
+  UnexpectedError,
+} from '../validators/errors';
+
+class EquationService {
+  async insertEquation(equation) {
+    try {
+      return await equationRepository.addEquation(equation);
+    } catch (error) {
+      if (error instanceof UniqueConstraintError) {
+        throw new CustomUniqueConstraintError(error.message);
+      }
+      if (error instanceof ValidationError) {
+        throw new CustomValidationError(error);
+      }
+
+      throw new UnexpectedError();
+    }
+  }
+
+  async getRandomEquation() {
+    try {
+      return await equationRepository.getRandomEquation();
```

Comment:
You don't need `return await` here. You can `await` it before you use it.

Reply:
but shouldn't I await to catch an error, if there exists one, and throw a customError instead of the original one?

Reply:
In this case i think this method is pretty useless and we should think a little more how to avoid these cases. Leave it like that since here error handling is happening.